controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -363,36 +363,45 @@ func TestHasRebootAnnotation(t *testing.T) {
 	host := newDefaultHost(t)
 	host.Annotations = make(map[string]string)
 
-	if hasRebootAnnotation(host) {
+	// FIXME (rdo) Here we ignore the isHardReboot boolean from
+	// hasRebootAnnotation(). We may want to expand the unit-test
+	// to validate the hard poweroff functionality as we build
+	// out such capabilities.
+	hasReboot, _ := hasRebootAnnotation(host)
```

Comment:
Instead of ignoring the hardMode value, I'd do e.g:

```
hasReboot, hardMode := hasRebootAnnotation(host)
assert.False(t, hasReboot)
assert.False(t, hardMode)
```

Then you can also add some new cases that cover the case where `hardMode=true`

Reply:
Thanks @hardys, I've added a new test case for the hard/soft modes and shifted the original test case to utilising assert.